### ISSUE TITLE
Why I have mbox_loss = 0 (* 1 = 0 loss) or nan in training?

### ISSUE THREAD

**Issue by nyanmn** [weiliu89/caffe#852]
I train my own dataset and I always have  mbox_loss = 0 (* 1 = 0 loss) or NAN.
I have all default parameters except my own image has 1280x720 size and I set

```
resize_width = 512
resize_height = 512
base_lr = 0.0004
batch_size = 16
accum_batch_size = 16
```

The output is like 
```
I0306 14:54:21.490119 17025 solver.cpp:243] Iteration 490, loss = nan
I0306 14:54:21.490231 17025 solver.cpp:259]     Train net output #0: mbox_loss = 0 (* 1 = 0 loss)

```

What could be wrong?

**Comment by LsFlyt** [comment]
May be your data is float.
but it only can read int.
such as
	<bndbox>
		<xmin>592.75</xmin>
		<ymin>378.0</ymin>
		<xmax>752.0</xmax>
		<ymax>541.0</ymax>
	</bndbox>
need to be
	<bndbox>
		<xmin>592</xmin>
		<ymin>378</ymin>
		<xmax>752</xmax>
		<ymax>541</ymax>
	</bndbox>

**Comment by nyanmn** [comment]
No my labels are in this format.
class_index     xmin                       ymin                     xmax                            ymax
0                     0.045833333333    0.2875                 0.163888888889        0.406944444444

**Comment by whcjb** [comment]
I think it is caused by small batch size, try on more GPUs and larger batch size.

**Comment by BOBrown** [comment]
mbox_loss = 0 stands for no positive examples are selected in your batch_size setting. Try increase batch size or decrease the overlap.

**Comment by HappyKerry** [comment]
@BOBrown  where to change the overlap value? Thanks

**Comment by ramgo2** [comment]
I was getting a similar error,
Train net output #0: mbox_loss = 0 (* 1 = 0 loss)

**Solution**: In my case, the labelmap had to be changed.. I had 5 classes with 0 being set to background and 1 to 4 for the other positive classes.  When I changed background id to 4 and changed the positive classes as 0 to 3 the mbox_loss was no longer 0.

Before getting to the solution, I checked the following (others had found success with one or more of the below changes),

- label format was fine.. there were no decimals or floating point (this post)

- Images weren't corrupt

- Padding images on all sides didn't help.

- Increasing batch size

- Random values in training data

- Reduce learning rate

https://github.com/sanghoon/pva-faster-rcnn/issues/8
https://github.com/MhLiao/TextBoxes/issues/14
https://www.bountysource.com/issues/55849760-why-i-have-mbox_loss-0-1-0-loss-or-nan-in-training
https://groups.google.com/forum/#!topic/caffe-users/vPZUcaooed4
https://stackoverflow.com/questions/49125030/why-i-have-mbox-loss-0-1-0-loss-or-nan-in-training
https://groups.google.com/forum/#!topic/caffe-users/Ns4thW5Y8BE
